frontend/src/components/elements/PlotlyChart/CustomTheme.tsx

```diff
@@ -426,55 +197,178 @@ export function applyStreamlitThemeTemplateLayout(
+    spec = spec.replaceAll(CATEGORY_6, "#FF6692")
+    spec = spec.replaceAll(CATEGORY_7, "#B6E880")
+    spec = spec.replaceAll(CATEGORY_8, "#FF97FF")
+    spec = spec.replaceAll(CATEGORY_9, "#FECB52")
+  }
+  return spec
 }
 
-/**
- * This applies specific changes for spec.layout.template.data because
- * spec.data does not fully catch these and neither does spec.layout.template.layout
- * @param data - spec.layout.template.data
- * @param theme - Theme from useTheme()
- */
-export function applyStreamlitThemeTemplateData(
-  data: any,
-  theme: Theme
-): void {
-  if (data !== undefined) {
-    data.bar.forEach((entry: any) => {
-      if (entry.marker !== undefined && entry.marker.line !== undefined) {
-        entry.marker.line = assign(entry.marker.line, {
-          color: getGray30(theme),
-        })
-      }
-    })
-    const graphs = Object.values(data)
-    graphs.forEach((graph: any) => {
-      if (graph.colorbar !== undefined && graph.colorbar.ticks === "") {
-        // make tick values show
-        delete graph.colorbar.ticks
-      }
-    })
+function replaceSequentialColors(
+  spec: string,
+  theme: Theme,
+  elTheme: string
+): string {
+  const SEQUENTIAL_0 = "#000011"
+  const SEQUENTIAL_1 = "#000012"
+  const SEQUENTIAL_2 = "#000013"
+  const SEQUENTIAL_3 = "#000014"
+  const SEQUENTIAL_4 = "#000015"
+  const SEQUENTIAL_5 = "#000016"
+  const SEQUENTIAL_6 = "#000017"
+  const SEQUENTIAL_7 = "#000018"
+  const SEQUENTIAL_8 = "#000019"
+  const SEQUENTIAL_9 = "#000020"
+
+  if (elTheme === "streamlit") {
+    const sequentialColors = getSequentialColorsArray(theme)
+    spec = spec.replaceAll(SEQUENTIAL_0, sequentialColors[0])
+    spec = spec.replaceAll(SEQUENTIAL_1, sequentialColors[1])
+    spec = spec.replaceAll(SEQUENTIAL_2, sequentialColors[2])
+    spec = spec.replaceAll(SEQUENTIAL_3, sequentialColors[3])
+    spec = spec.replaceAll(SEQUENTIAL_4, sequentialColors[4])
+    spec = spec.replaceAll(SEQUENTIAL_5, sequentialColors[5])
+    spec = spec.replaceAll(SEQUENTIAL_6, sequentialColors[6])
+    spec = spec.replaceAll(SEQUENTIAL_7, sequentialColors[7])
+    spec = spec.replaceAll(SEQUENTIAL_8, sequentialColors[8])
+    spec = spec.replaceAll(SEQUENTIAL_9, sequentialColors[9])
+  } else {
+    // Default plotly colors
+    spec = spec.replaceAll(SEQUENTIAL_0, "#0d0887")
+    spec = spec.replaceAll(SEQUENTIAL_1, "#46039f")
+    spec = spec.replaceAll(SEQUENTIAL_2, "#7201a8")
+    spec = spec.replaceAll(SEQUENTIAL_3, "#9c179e")
+    spec = spec.replaceAll(SEQUENTIAL_4, "#bd3786")
+    spec = spec.replaceAll(SEQUENTIAL_5, "#d8576b")
+    spec = spec.replaceAll(SEQUENTIAL_6, "#ed7953")
+    spec = spec.replaceAll(SEQUENTIAL_7, "#fb9f3a")
+    spec = spec.replaceAll(SEQUENTIAL_8, "#fdca26")
+    spec = spec.replaceAll(SEQUENTIAL_9, "#f0f921")
   }
+  return spec
+}
+
+function replaceDivergingColors(
+  spec: string,
+  theme: Theme,
+  elTheme: string
+): string {
+  const DIVERGING_0 = "#000021"
```

Comment:
nit: this currently uses different constant names compared to the python version (DIVERGING_0 vs DIVERGING_20). This should be probably unified, I think I would prefer the numbering starting from 0 slightly more

Reply:
and maybe add a comment to make it clear that these constants are matching the placeholder constants used in the python backend, e.g.:

// All the placeholder constants defined here are matching the placeholders in the python implementation.